buildcheck/backend/visualizer.py

```diff
@@ -1,442 +1,335 @@
-        self._draw_ocr_results(ax, ocr_processor, image_path)
-        
-        # Add OCR-specific legend
-        self._add_ocr_legend(ax)
-        
-        ax.set_title('Floor Plan OCR Analysis', fontsize=16, fontweight='bold')
-        ax.axis('off')
-        
-        plt.tight_layout()
-        
+from buildcheck.backend.vectorization import *
+from buildcheck.backend.ocr_processor import OCRProcessor
+from buildcheck.backend.yolo_processor import YOLOProcessor
 
-        
-        plt.show()
+
+class FloorPlanVisualizer:
+    def __init__(self, image_path: str, model_path: str = None):
+        self.image_path = image_path
+        self.model_path = model_path
+        self.layout = None
+        self.room_colors = {}
+        self.room_names = []
+        
+    def generate_distinct_colors(self, n: int) -> List[Tuple[float, float, float]]:
+        """Generate n visually distinct colors"""
+        colors = []
+        for i in range(n):
+            hue = i / n
+            saturation = 0.3 + (i % 2) * 0.3  # Alternate between 0.3 and 0.6
+            value = 0.8 + (i % 3) * 0.1  # Vary brightness slightly
+            rgb = colorsys.hsv_to_rgb(hue, saturation, value)
+            colors.append(rgb)
+        return colors
     
-    def visualize_yolo(self, image_path: str, layout: Layout,
-                      show_rooms: bool = True,
-                      save_path: str = None):
-        """
-        Visualize YOLO detection results on the floor plan image.
-        
-        Args:
-            image_path: Path to the floor plan image
-            layout: Layout object containing rooms and symbols
-            show_rooms: Whether to show room boundaries
-            save_path: Path to save the visualization (optional)
-        """
-        
-        # Load and display the image
-        image = cv2.imread(image_path)
-        image_rgb = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+    def process_floor_plan(self):
+        """Process the floor plan with both OCR and YOLO"""
+        # Create layout with manual room boundaries
+        self.layout = create_test_layout()
         
-        fig, ax = plt.subplots(1, 1, figsize=(15, 12))
-        ax.imshow(image_rgb)
+      
         
-        # Show room boundaries
-        if show_rooms:
-            self._draw_room_boundaries(ax, layout)
         
-        # Show YOLO results
-        self._draw_yolo_results(ax, layout)
+        # Process OCR
+        print("Processing OCR...")
+        ocr_processor = OCRProcessor(self.image_path, self.layout)
+        ocr_processor.ocrProcess()
         
-        # Add YOLO-specific legend
-        self._add_yolo_legend(ax, layout)
-        
-        ax.set_title('Floor Plan YOLO Detection Analysis', fontsize=16, fontweight='bold')
-        ax.axis('off')
-        
-        plt.tight_layout()
-        
-
-        
-        plt.show()
+        # Process YOLO if model path is provided
+        if self.model_path:
+            print("Processing YOLO...")
```

Comment:
YOLO is a required part of our software. there is no scenario where we should allow not running yolo. let's disallow the default None value and simply raise an exception if model_path is invalid